jsonrpc/dispatcher.go

```diff
@@ -231,54 +237,90 @@ func (d *Dispatcher) RemoveFilterByWs(conn wsConn) {
 }
 
 func (d *Dispatcher) HandleWs(reqBody []byte, conn wsConn) ([]byte, error) {
-	var req Request
-	if err := json.Unmarshal(reqBody, &req); err != nil {
-		return NewRPCResponse(req.ID, "2.0", nil, NewInvalidRequestError("Invalid json request")).Bytes()
-	}
+	const (
+		openSquareBracket  byte = '['
+		closeSquareBracket byte = ']'
+		comma              byte = ','
+	)
 
-	// if the request method is eth_subscribe we need to create a
-	// new filter with ws connection
-	if req.Method == "eth_subscribe" {
-		filterID, err := d.handleSubscribe(req, conn)
-		if err != nil {
-			return NewRPCResponse(req.ID, "2.0", nil, err).Bytes()
-		}
+	reqBody = bytes.TrimLeft(reqBody, " \t\r\n")
 
-		resp, err := formatFilterResponse(req.ID, filterID)
+	// if body begins with [ than consider this request as batch request
+	if len(reqBody) > 0 && reqBody[0] == openSquareBracket {
+		var batchReq BatchRequest
 
+		err := json.Unmarshal(reqBody, &batchReq)
 		if err != nil {
-			return NewRPCResponse(req.ID, "2.0", nil, err).Bytes()
+			return NewRPCResponse(nil, "2.0", nil,
```

Comment:
Should we provide request id as part of `RPCResponse` even in case of returning an error?